### ISSUE TITLE
API support for bulk object updates/edits (put/patch)

### ISSUE THREAD

**Issue by ttarabula** [netbox-community/netbox#4882]
### Environment
* Python version:  Python 3.6
* NetBox version:  Netbox 2.8.8

### Proposed Functionality

Allow for bulk editing (put/patch) of devices (preferably any object) through API end-points.

Related issue (bulk deletes): #3436 

### Use Case

Currently, iterative updates to objects doesn't scale well and API support for a single call to edit multiple objects via the API could streamline various access patterns. More generally though, since bulk create (post) is already supported, and the bulk delete enhancement proposal was accepted, it would be nice to have this as well for completeness' sake. Similar to bulk delete, since the web interface also supports this, I can see official support in the API simplifying the web interface code as well.

**Comment by jeremystretch** [comment]
Marking as blocked by #3436 (still need to address the schema generation issue)
